flask_mwoauth/__init__.py

```diff
@@ -131,31 +137,38 @@ def request(self, api_query, url=None):
             # see https://www.mediawiki.org/wiki/API:Edit#Large_texts) then
             # transmit as multipart message
 
-            req = self._prepare_long_request(url=url + "/api.php?",
-                                             api_query=api_query
-                                             )
-            return self.mwoauth.post(url + "/api.php?",
-                                     data=req.body,
-                                     content_type=req.headers['Content-Type']
-                                     ).data
+            req = self._prepare_long_request(url=api_url,
+                                             api_query=api_query)
+            req.send()
+            return req.response.text
         else:
-            return self.mwoauth.post(url + "/api.php",
-                                     data=api_query
-                                     ).data
+            return requests.post(api_url, data=api_query).text
 
     def get_current_user(self, cached=True):
         if cached:
-            return session.get('username')
-
-        try:
-            data = self.request({'action': 'query', 'meta': 'userinfo'})
-            session['username'] = data['query']['userinfo']['name']
-        except KeyError:
-            session['username'] = None
-            if data['error']['code'] == "mwoauth-invalid-authorization":
-                flash(u'Access to this application was revoked. Please re-login!')
-            else:
-                raise
-        except OAuthException:
-            session['username'] = None
-        return session['username']
+            return session.get('mwoauth_username')
+
+        # Get user info
+        identity = self.handshaker.identify(
+            mwoauth.AccessToken(**session['mwoauth_access_token']))
```

Comment:
Breaking change: calling `get_current_user(cached=False)` on unlogged users raises `KeyError` from here instead of returning `None`. Is this intended?